Add media selector to picture-card-editor

karwosts opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Proposed change
Add a media selector to picture-card-editor, which launches the media browser dialog and lets you pick a media image. 

(redacted original image, see thread for new proposal)

## Type of change
<!--
  What type of change does your PR introduce to the Home Assistant frontend?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration
<!--
  Supplying a configuration snippet, makes it easier for a maintainer to test
  your PR.
-->

```yaml

```

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

<!--
  Thank you for contributing <3
-->

[docs-repository]: https://github.com/home-assistant/home-assistant.io

karwosts committed:
Add media selector to picture card

wendevlin commented:
Awesome, I'll request help from UX here.

I would not show the media path input, moreover I would maybe remove the "select from media" in the image upload.

wendevlin commented:
> We can hide the media path here under the button, but without that there currently isn't any feedback in the control that what you've selected has changed anything, the 'pick media' button itself doesn't change after you've selected something. Of course if everything is working properly you can see the change in the right hand side of the card editor, but given this change is in a selector we might also consider how it could be reused in other contexts without that card preview.

You are right we should provide some klind of feedback. But when we add a UI here we should then also provide a UI to show what was selected and if the selection was successful, instead of the same manual field we already had. That could lead to confusions. 


> > I would maybe remove the "select from media" in the image upload
> 
> I'm not sure I see the harm of leaving that in there, since it's tied directly into the "uploads" folder. So maybe it make sense when you're picking from uploads, to have that link to keep all the upload feature together? Otherwise you have to know where to find your uploads starting from the main media root folder. I could go either way on this, just thinking about reasons why it could make sense to leave it.

That is true, it's more context focus, let's keep it :+1:

marcinbauer85 commented:
Hey @karwosts 👋 
I think we already have this in the `<ha-selector-background>`, which is used in edit -> view:
1. Dropzone with pick local existing file (Picture card option 1 and 2)
2. Local path or web URL -> With this for view background edit, we decided to move the URL to yaml mode, not to introduce breaking changes, but still have the ability for advanced users, since an external image can pose privacy and security reasons.
I'd go with just using that component.
Side note: The component name is too specific and could be renamed `<ha-selector-image>` to fit more cases.

marcinbauer85 commented:
@karwosts Could the `<ha-selector-image>` just get the ability that you've added here? The UI that is used in the View / Background (tab) configuration is simple and provides all of the functionalities:
1. Adding (uploading) images
2. Selecting previously uploaded images via media browser (only "Image Upload" folder, but could be expanded to other types in other usecases, like Picture card?)
3. Clearing an image
4. Setting a URL (local or remote) via YAML

karwosts commented:
I think you're mixing up some terms.

`<ha-selector-image>` is what's currently used by all the image-related frontend card editors, core services or blueprints that use an image selector, and what I've added this capability to.

The element in the view background editor is `<ha-selector-background>`, and is used only there. 

I think what you're saying is you want to replace all the image-card editors with the design from ha-selector-background.

However I am not in favor of removing the URL path textbox from the card editors. That seems user-hostile to me. The default card stub uses the URL path, and probably 99% of the current use of the selector still uses the URL path. Forcing to go to yaml mode for that, I don't understand that.

I think we maybe had the same debate last time I added the upload dialog to the picture card editor, and we settled on this design with the radio buttons.

karwosts committed:
revert previous changes

karwosts committed:
Merge branch 'dev' of https://github.com/home-assistant/frontend into picture-card-media-selector

karwosts committed:
everything in media selector

karwosts commented:
Ok I've tried to get all this functionality into the media selector, which required merging some of the functions of the media selector with the picture-upload module.

This is what the default stub config now looks like:

<img width="1003" height="613" alt="image" src="https://github.com/user-attachments/assets/34dff83e-89c2-45c8-b186-d0362e8bf77e" />

You can:
 1. Click the media-selector, to open the select dialog to browse in the media root, to select any of the other media items, or use `Manual entry` to edit the URL to a different URL.
 2. Click the "Clear picture" button to erase the current media content. When the media is completely cleared, the selector changes to this view of a picture upload.
 
 
<img width="1013" height="759" alt="image" src="https://github.com/user-attachments/assets/b4912b62-4790-4c11-8c27-d433b35d5796" />

With the picture upload, you can drag/drop or browse an image which will be uploaded to image upload. Or click the 'select from media' link to again get the root media browse dialog, to select an pre-existing media.

If a new upload is performed, that media is then selected, and it looks again like the first view:
<img width="1007" height="294" alt="image" src="https://github.com/user-attachments/assets/ac66fe62-e87a-432c-9081-1f2813f99837" />

Clicking the selector here opens the browse dialog in the image upload folder, so you can pick another upload, or go up a directory to get back to the root browse, or select a different media content.

karwosts committed:
lint

karwosts committed:
comment

karwosts committed:
Properties to customize the id helper and hide content type

karwosts committed:
Merge branch 'dev' into picture-card-media-selector

karwosts committed:
Merge branch 'dev' into picture-card-media-selector

karwosts committed:
merge/ci fix

wendevlin reviewed (approved):
Thanks 🙂
